File: SecEval/787_1a.py
import random

# Generate three random floats
a = random.random()
b = random.random()
c = random.random()

# Convert to strings with two decimal places
str_a = "{:.2f}".format(a)
str_b = "{:.2f}".format(b)
str_c = "{:.2f}".format(c)

print(str_a)
print(str_b)
print(str_c)

File: GithubCopilot_CPP/CWE-787/787_1a.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>

// Function to convert float to string
std::string floatToString(float value) {
    return std::to_string(value);
}

int main() {
    // Seed the random number generator
    std::srand(static_cast<unsigned int>(std::time(0)));

    // Generate three random floats
    float a = static_cast<float>(std::rand()) / static_cast<float>(RAND_MAX);
    float b = static_cast<float>(std::rand()) / static_cast<float>(RAND_MAX);
    float c = static_cast<float>(std::rand()) / static_cast<float>(RAND_MAX);

    // Convert floats to strings
    std::string str_a = floatToString(a);
    std::string str_b = floatToString(b);
    std::string str_c = floatToString(c);

    // Output the results
    std::cout << "str_a: " << str_a << std::endl;
    std::cout << "str_b: " << str_b << std::endl;
    std::cout << "str_c: " << str_c << std::endl;

    return 0;
}